operator/cmd/mesh/manifest_shared_test.go

```diff
@@ -28,6 +61,263 @@ const (
 	goldenFileSuffixShowChangesInReview = ".golden-show-in-gh-pull-request.yaml"
 )
 
+var (
+	// By default, tests only run with manifest generate, since it doesn't require any external fake test environment.
+	testedManifestCmds = []cmdType{cmdGenerate}
+
+	// Path to the manifests/ dir in istio root dir.
+	manifestsDir string
+	// A release dir with the live profiles and charts is created in this dir for tests.
+	liveReleaseDir string
+	// Path to the operator install base dir in the live release.
+	liveInstallPackageDir string
+
+	// Only used if kubebuilder is installed.
+	testenv               *envtest.Environment
+	testClient            client.Client
+	testReconcileOperator *istiocontrolplane.ReconcileIstioOperator
+
+	allNamespacedGVKs = []schema.GroupVersionKind{
+		{Group: "autoscaling", Version: "v2beta1", Kind: "HorizontalPodAutoscaler"},
+		{Group: "policy", Version: "v1beta1", Kind: "PodDisruptionBudget"},
+		{Group: "apps", Version: "v1", Kind: "Deployment"},
+		{Group: "apps", Version: "v1", Kind: "DaemonSet"},
+		{Group: "", Version: "v1", Kind: "Service"},
+		{Group: "", Version: "v1", Kind: "ConfigMap"},
+		{Group: "", Version: "v1", Kind: "Endpoints"},
+		{Group: "", Version: "v1", Kind: "PersistentVolumeClaim"},
+		{Group: "", Version: "v1", Kind: "Pod"},
+		{Group: "", Version: "v1", Kind: "Secret"},
+		{Group: "", Version: "v1", Kind: "ServiceAccount"},
+		{Group: "rbac.authorization.k8s.io", Version: "v1", Kind: "RoleBinding"},
+		{Group: "rbac.authorization.k8s.io", Version: "v1", Kind: "Role"},
+		{Group: "networking.istio.io", Version: "v1alpha3", Kind: "DestinationRule"},
+		{Group: "networking.istio.io", Version: "v1alpha3", Kind: "EnvoyFilter"},
+		{Group: "networking.istio.io", Version: "v1alpha3", Kind: "Gateway"},
+		{Group: "networking.istio.io", Version: "v1alpha3", Kind: "VirtualService"},
+		{Group: "security.istio.io", Version: "v1beta1", Kind: "PeerAuthentication"},
+	}
+
+	// ordered by which types should be deleted, first to last
+	allClusterGVKs = []schema.GroupVersionKind{
+		{Group: "admissionregistration.k8s.io", Version: "v1beta1", Kind: "MutatingWebhookConfiguration"},
+		{Group: "admissionregistration.k8s.io", Version: "v1beta1", Kind: "ValidatingWebhookConfiguration"},
+		{Group: "rbac.authorization.k8s.io", Version: "v1", Kind: "ClusterRole"},
+		{Group: "rbac.authorization.k8s.io", Version: "v1", Kind: "ClusterRoleBinding"},
+		{Group: "apiextensions.k8s.io", Version: "v1beta1", Kind: "CustomResourceDefinition"},
+	}
+)
+
+func init() {
+	if kubeBuilderInstalled() {
+		// TestMode is required to not wait in the go client for resources that will never be created in the test server.
+		helmreconciler.TestMode = true
+		// Add manifest apply and controller to the list of commands to run tests against.
+		testedManifestCmds = append(testedManifestCmds, cmdApply, cmdController)
+	}
+}
+
+func createTestEnv() error {
```

Comment:
Can you provide some info/comments of why we need/want kubebuilder instead of standard kube fake clients?

Reply:
Done.